league_rpc_linux/processes/lcu_thread.py

```diff
@@ -0,0 +1,377 @@
+import time
+from threading import Timer
+from dataclasses import dataclass
+
+import lcu_driver
+from lcu_driver.connection import Connection
+from lcu_driver.events.responses import WebsocketEventResponse
+from pypresence import Presence
+
+from league_rpc_linux.colors import Colors
+from league_rpc_linux.const import (
+    BASE_MAP_ICON_URL,
+    GAME_MODE_CONVERT_MAP,
+    LEAGUE_OF_LEGENDS_LOGO,
+    MAP_ICON_CONVERT_MAP,
+    PROFILE_ICON_BASE_URL,
+    SMALL_TEXT,
+)
+
+
+@dataclass
+class ClientData:
+    availability: str = "Online"  # "Online", "Away
+    gamemode: str = None
+    gameflow_phase: str = "None"  # None, Lobby, Matchmaking, ReadyCheck, ChampSelect, InProgress, WaitingForStats, EndOfGame
+    is_custom: bool = False
+    is_practice: bool = False
+    lobby_id: str = None  # unique lobby id
+    map_id: int = None  # 11, 12, 21, 22, 30
+    max_players: int = None  # max players in lobby
+    players: int = None  # players in lobby
+    queue: str = None
+    queue_id: int = -1
+    queue_is_ranked: bool = False
+    summoner_icon: int = None
+    summoner_id: str = None
+    summoner_level: int = None
+    summoner_name: str = None
+    summoner_rank: str = None
+
+
+## Timers ##
+
+
+# As some events are called multiple times, we should limit the amount of updates to the RPC.
+# Collect update events for 1 second and then update the RPC.
+class RPCUpdater:
+    def __init__(self):
+        self.scheduled_update = False
+
+    def delay_update(self):
+        if not self.scheduled_update:
+            self.scheduled_update = True
+            Timer(1.0, self.update_rpc_and_reset_flag).start()
+
+    def update_rpc_and_reset_flag(self):
+        module_data.update_rpc()  # Assuming update_rpc is defined elsewhere
```

Comment:
the type of module_data is unknown. because the class RPCUpdater has no idea if an instance, called "module_data" is anywhere in this scope. 

Consider sending in the module_data as a parameter for "update_rpc_and_reset_flag" and type hint it correctly, so that it is aware of the function it's about to run.

The easiest way would be to split all of the different functionalities in this file, into separate modules and then import them all here.. we don't have to have the update_rpc inside of a class. i think.

I will test it myself and see if i can get something done, and perhaps send a PR to this branch

Reply:
Please check latest commit. This comment has been fixed.